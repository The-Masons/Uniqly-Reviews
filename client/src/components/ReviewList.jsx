import React from 'react';
import ReactDOM from 'react-dom';
import axios from 'axios';
import Review from './Review.jsx';
import OverallRatings from './OverallRatings.jsx';

class ReviewList extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            productData: [],
            product_id: null,
        }
    }

    componentDidMount() {
        const context = this;
        let url = window.location.href.split('/').pop();
        (url.length > 0) ? null : url = '0';
        axios.get('/reviews/' + url)
            .then(function (response) {
                context.setState({
                    productData: response.data,
                    product_id: url
                });
            })
            .catch(function (error) {
                console.log(error);
            })
    }

    render() {
        return (
            <div className='reviews'>
                <div className='overall-ratings'>
                    <OverallRatings reviewData={this.state.productData} />
                </div>
                {this.state.productData.map((item, index) =>
                    <div key={index}>
                        <Review reviewData={this.state.productData[index]} />
                    </div>)
                }
            </div>
        );
    }
}

export default ReviewList;
