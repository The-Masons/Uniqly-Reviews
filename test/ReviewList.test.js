import React from 'react';
import ReactDOM from 'react-dom';
import ReviewList from '../client/src/components/ReviewList.jsx';
import renderer from 'react-test-renderer';

const sample = {
  "product_id": 0,
  "nickname": "Lisa",
  "from": "East Gladysville, Hawaii",
  "size_purchased": "XS",
  "age": "18 to 24",
  "gender": "F",
  "received_free_product": false,
  "chest_size": 45,
  "waist_size": 35,
  "height": "5'8",
  "hips_size": 34,
  "product_fit": 5,
  "product_length": 1,
  "product_quality": 3,
  "date": "2017-05-09T22:03:51.414Z",
  "rating": 5,
  "review_title": "corporis iure molestiae",
  "review_body": "Non a inventore magnam corporis et voluptas doloribus reiciendis autem. Ut iusto ipsum explicabo sunt quod. Ab quis optio provident magnam similique eligendi quia unde sunt. Velit odit magnam. Tempore recusandae aliquid labore optio excepturi sed voluptas eveniet.",
  "recommended": false,
  "helpful": 13,
  "from_uk": true,
  "verified_purchaser": false,
  "employee": false
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ReviewList reviewData={sample}/>, div);
});

describe('ReviewList', () => {
  test('should render with the correct items', () => {
    const itemPicker = mount(
      <ReviewList
        reviewData={sample}/>
      );
    expect(ReviewList).toMatchSnapshot();
  });
});