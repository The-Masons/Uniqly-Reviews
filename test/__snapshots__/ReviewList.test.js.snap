// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList should render with the correct items 1`] = `[Function]`;
